gradebookng/tool/src/webapp/scripts/gradebook-gbgrade-table.js

```diff
@@ -191,6 +193,30 @@ TrimPathFragmentCache.prototype.setHTML = function (target, values) {
   GbGradeTable.replaceContents(target, this.getFragment(values));
 }
 
+GbGradeTable.matchesItemFilter = function(column) {
+  if (!GbGradeTable.itemFilterText) {
+    return true;
+  }
+
+  const haystack = [
+    column?.title,
+    column?.abbrevTitle,
+    column?.name,
+    column?.categoryName,
+  ].filter(Boolean).join(" ").toLowerCase();
+
+  return haystack.indexOf(GbGradeTable.itemFilterText) !== -1;
+};
+
+GbGradeTable.isColumnHidden = function(column) {
+  return column.hidden || !GbGradeTable.matchesItemFilter(column);
+};
+
+GbGradeTable.setItemFilterText = function(text) {
+  GbGradeTable.itemFilterText = (text || "").toString().trim().toLowerCase();
+  GbGradeTable.redrawTable(true);
+};
+
```

Comment:
i think this needs to more optimized for perfromance. performance is very important in GB. how many times it would run if there are more than 50 columns.